controllers/front/CartController.php

```diff
@@ -349,20 +396,52 @@ protected function processChangeProductInCart()
             }
 
             // Check customizable fields
-            if ($product->hasActivatedRequiredCustomizableFields() && !$this->customization_id) {
-                $this->errors[] = $this->trans('Please fill in all of the required fields, and then save your customizations.', array(), 'Shop.Notifications.Error');
+
+            if (!$product->hasAllRequiredCustomizableFields() && !$this->customization_id) {
+                array_push(
+                    $this->{$ErrorKey},
+                    $this->trans(
+                        'Please fill in all of the required fields, and then save your customizations.',
+                        array(),
+                        'Shop.Notifications.Error'
+                    )
+                );
             }
 
             if (!$this->errors) {
                 $cart_rules = $this->context->cart->getCartRules();
                 $available_cart_rules = CartRule::getCustomerCartRules($this->context->language->id, (isset($this->context->customer->id) ? $this->context->customer->id : 0), true, true, true, $this->context->cart, false, true);
-                $update_quantity = $this->context->cart->updateQty($this->qty, $this->id_product, $this->id_product_attribute, $this->customization_id, Tools::getValue('op', 'up'), $this->id_address_delivery);
+                $update_quantity = $this->context->cart->updateQty(
+                    $this->qty,
+                    $this->id_product,
+                    $this->id_product_attribute,
+                    $this->customization_id,
+                    Tools::getValue('op', 'up'),
+                    $this->id_address_delivery,
+                    null,
+                    true,
+                    true
+                );
                 if ($update_quantity < 0) {
                     // If product has attribute, minimal quantity is set with minimal quantity of attribute
                     $minimal_quantity = ($this->id_product_attribute) ? Attribute::getAttributeMinimalQty($this->id_product_attribute) : $product->minimal_quantity;
-                    $this->errors[] = $this->trans('You must add %d minimum quantity', array($minimal_quantity), 'Shop.Notifications.Error');
+                    array_push(
+                        $this->{$ErrorKey},
+                        $this->trans(
+                            'You must add %d minimum quantity',
+                            array($minimal_quantity),
```

Comment:
You can fix an issue at the same time here. All variables to apply on a translated string must be associated to a placeholder.

```php
                        $this->trans(
                            'You must add %quantity% minimum quantity',
                            array('%quantity%' => $minimal_quantity),
                            [...]
```

Reply:
Done :)